CONTRIBUTING.md

```diff
@@ -1,172 +1,120 @@
-
-If you want to update the data:
-
-The entry point is at [data/Main.scala](/data/src/main/scala/scaladex/data/Main.scala)
-
-### List Poms
-
-This step searches on Bintray for artifact containing `_2.10`, `_2.11`, `_2.12`, ...
-Bintray contains jcenter, it's a mirror of maven central.
-
-You will need a premium Bintray account.
-
+2. Then you can start the server with:
 ```
-set javaOptions in reStart ++= Seq(
-  "-DBINTRAY_USER=XXXXXXX", 
-  "-DBINTRAY_PASSWORD=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-)
-
-data/reStart list
+sbt:scaladex> server/run
 ```
+3. Finally you can open the website in your browser at `localhost:8080`.
 
-### Download ivy.xml of SBT plugins from Bintray
-
-This step also requires a premium Bintray.
+The server will start fetching information from Github progressively, you will see them appear after some time.
 
-```
-data/reStart sbt
-```
+After you stop and start the server again, the github information will still be there.
 
-### Download Poms
+## Organization of the code
 
-This steps download poms and parent poms from Bintray
+The build contains 6 modules (there are all under the `modules/` directory):
+- `core`: 
+it contains the core data classes and service interfaces of Scaladex.
+It is cross-compiled into JVM bytecode for the `server` and Javascript for the `webclient`.
+- `data`:
+Some useful operations to update the Scaladex data.
+Only the `init` operation is still used, the other operations are progressively translated to scheduled jobs in the `server` module.
+- `infra`:
+It contains the implementations of the all services, using PostgreSQL for the storage, Elasticsearch for the search engine and akka-http for the third-party APIs (Github, Maven Central...).
+- `server`:
+The Scaladex server.
+- `template`:
+The HTML templates of the pages that are generated by the server.
+- `webclient`:
+The script compiled to Javascript and executed by the browser.
 
-```
-data/reStart download
-data/reStart parent
-```
+## Architecture
 
-### GitHub
+Scaladex receives POM files from Maven Central, it parses them, find the Github URL of the project and stores the information in the `artifacts` and `projects` table of the database.
 
-This step downloads GitHub metadata and content.
+In the Scaladex terminology, an artifact is the POM file of a Scala artifact and a project is the Github repository of a Scala project.
+An artifact must be associated to a single project.
+A project can have many artifacts.
 
-You need to create a token on your github account: https://github.com/settings/tokens/new
+Locally, we do not receive any POM files, instead we use the `data / run init` to populate the database with some initial artifacts and projects.
 
-Then fill the following file `../scaladex-dev-credentials/application.conf` with:
+While running, the server executes some scheduled jobs periodically to update the information of the projects and to synchronize the search engine:
```

Comment:
It makes think that locally, developper wont' be admin.. maybe we should add isAdmin = env.isLocal or part of scala center

Reply:
Oh yes that's a good idea